### PR TITLE
ButtonGroup POC

### PR DESCRIPTION
**Pull request by jonnystening**
@simonswiss Is this a better approach to child styling, rather than overriding child button styling via `:first-child`/`:last-child` CSS selectors?

### CODE REVIEW

**Comment by jonnystening** [comment]
After discussing with @simonswiss, we are best to make use of CSS overrides to provide this first and last child styling, rather than invoking the help of cloneElement() to pass a first/last prop and style based on that.

**Comment by jonnystening** [comment]
...then this came across my desk: https://github.com/WestpacGEL/GEL/pull/148#discussion_r331280867

Hmm 🤔, perhaps the use of cloneElement() in this case (to pass first/last props), is kosher after all?

File: components/button-group/src/ButtonGroup.js [hunk 1]
```diff
@@ -37,6 +37,18 @@ export const ButtonGroup = ({
...
+	const childrenWithProps = Children.map(children, (child, index) => {
+		if (index === 0) {
+			return cloneElement(child, { first: true })
+		}
+		if (index === numChildren - 1) {
+			return cloneElement(child, { last: true })
+		}
+
+		return child;
+	});
```
Comment (simonswiss) [inline components/button-group/src/ButtonGroup.js]: As per our discussion + my overnight epiphany - this is actually a good use case for `cloneElement`.

Also, multi-line comments FTW!

Comment (jonnystening) [reply components/button-group/src/ButtonGroup.js]: 🤣

**Pull request merged**